pimd/pim_igmp.c

```diff
@@ -270,6 +270,23 @@ void pim_igmp_other_querier_timer_off(struct igmp_sock *igmp)
 	THREAD_OFF(igmp->t_other_querier_timer);
 }
 
+int igmp_validate_checksum(char *igmp_msg, int igmp_msg_len)
+{
+	uint16_t recv_checksum;
+	uint16_t checksum;
+
+	recv_checksum = *(uint16_t *)(igmp_msg + IGMP_CHECKSUM_OFFSET);
+
+	/* for computing checksum */
+	*(uint16_t *)(igmp_msg + IGMP_CHECKSUM_OFFSET) = 0;
```

Comment:
Convenient as this seems to be, you can't cast `char *` to `uint16_t *`. `uint16_t` must be 2-byte aligned.

Reply:
This was the existing code, I just moved it to a new function :) I will check it what can be done about it.

Reply:
This is fixed